src/app/main/services/recipes-service/recipes.service.ts

```diff
@@ -0,0 +1,58 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, Subscribable, Subscription, map, mergeAll, of, tap } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Recipe } from 'src/interfaces/recipe';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RecipesService {
```

Comment:
it looks like a service to work with http requests. so make a separate layer with services only task which is connect to api and return stream with data (it should be in some general folder to reuse it from some other services or components)
all other business logic like map this data should do separate layer which responsible for business logic (component if this logic not too large or other reusable service which can be near the component which use it)

Reply:
understood!
thanks